lbrynet/file_manager/EncryptedFileCreator.py

```diff
@@ -70,8 +63,27 @@ def _make_stream_hash(self):
 
     def _finished(self):
         self._make_stream_hash()
-        d = self._save_stream_info()
-        return d
+        return defer.succeed(self.stream_hash)
+
+    def get_sd_info(self, stream_type=EncryptedFileStreamType):
+        sd_blob = {}
+        sd_blob['stream_type'] = stream_type
+        sd_blob['stream_name'] = hexlify(self.name)
+        sd_blob['key'] = hexlify(self.key)
+        sd_blob['suggested_file_name'] = hexlify(self.suggested_file_name)
+        sd_blob['stream_hash'] = self.stream_hash
+
+        formatted_blobs = []
+        for blob_info in self.blob_infos:
+            blob = {}
+            if blob_info.length != 0:
+                blob['blob_hash'] = blob_info.blob_hash
+            blob['blob_num'] = blob_info.blob_num
+            blob['iv'] = blob_info.iv
+            blob['length'] = blob_info.length
+            formatted_blobs.append(blob)
+        sd_blob['blobs'] = formatted_blobs
+        return defer.succeed(sd_blob)
```

Comment:
I'm having a bit of déjà vu here, is this function defined already somewhere else? `lbrynet/core/StreamDescriptor.py` They may do something slightly different but the function names are identical and they appear to be very similar internally.

Reply:
now the formatting is only in `lbrynet/core/StreamDescriptor.py`